qiskit/execute_function.py

```diff
@@ -275,6 +302,182 @@ def execute(
+        warnings.warn(
+            "The memory_slot_size argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if rep_time is not None:
+        warnings.warn(
+            "The rep_time argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if rep_delay is not None:
+        warnings.warn(
+            "The rep_delay argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if not schedule_circuit:
+        warnings.warn(
+            "The schedule_circuit argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require scheduling your circuit "
+            "prior to execution you should use the :func:`~.schedule` function explicitly with "
+            "prior to running the circuit.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if inst_map is not None:
+        warnings.warn(
+            "The inst_map argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require compiling to a "
+            "custom :class:`~.InstructionScheduleMap` you should use the transpile() function "
+            "instead.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if meas_map is not None:
+        warnings.warn(
+            "The meas_map argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require scheduling your circuit "
+            "prior to execution you should use the :func:`~.schedule` function explicitly with "
+            "prior to running the circuit.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if scheduling_method is not None:
+        warnings.warn(
+            "The scheduling_method argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require scheduling your circuit "
+            "prior to execution you should use the :func:`~.schedule` function explicitly with "
+            "prior to running the circuit.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+    if init_qubits is not None:
+        warnings.warn(
+            "The init_qubits argument is deprecated as of Qiskit Terra 0.21.0, "
+            "and will be removed in a future release. If you require scheduling your circuit "
+            "prior to execution you should use the :func:`~.schedule` function explicitly with "
+            "prior to running the circuit.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
```

Comment:
Users with several arguments in `execute` will a lot get a lot of warning (maybe flooding the terminal?). I suggest this style:
```suggestion
    if scheduling_method is not None:
        deprecated_arg.append('init_qubits')

        if deprecated_arg:
        warnings.warn(
            "The following arguments are deprecated as of Qiskit Terra 0.21.0, "
            "and will be removed in a future release: {", ".join(deprecated_arg)}. If you require scheduling your circuit "
            "prior to execution you should use the :func:`~.schedule` function explicitly with "
            "prior to running the circuit.",
            DeprecationWarning,
            stacklevel=2,
        )
```